### PR TITLE
[DEV-2484] obs_table: persist definition when creating from view

### PR DESCRIPTION
**Pull request by jevonyeoh**
## Description
Persist the definition of how we create an observation table when it's created from a view. This will allow us to display it in the UI.

## Related Issue
https://featurebyte.atlassian.net/browse/DEV-2484

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: tests/unit/api/observation_table/test_utils.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    event_view = event_table.get_view(
+        view_mode="manual",
+        drop_column_names=["created_at"],
+        column_cleaning_operations=[],
+    )
+    output = event_view.create_observation_table(
+        name="test_obs_table",
+        sample_rows=100,
+        columns=["event_id", "event_name"],
+        columns_rename_mapping={{"event_id": "id"}},
```
Comment (YS-L) [inline tests/unit/api/observation_table/test_utils.py:44]: Is the double curly brackets correct?

File: tests/unit/api/observation_table/test_utils.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        context_name=test_context,
```
Comment (YS-L) [inline tests/unit/api/observation_table/test_utils.py:45]: `test_context` should be made a literal string.

File: tests/unit/api/observation_table/test_utils.py (hunk 1 above)
Comment (jevonyeoh) [reply tests/unit/api/observation_table/test_utils.py:44]: these are doubled here as escape characters

File: tests/unit/api/observation_table/test_utils.py (hunk 2 above)
Comment (jevonyeoh) [reply tests/unit/api/observation_table/test_utils.py:45]: ooh good catch

File: tests/unit/api/observation_table/test_utils.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        name="test_obs_table",
+        sample_rows=100,
+        columns=["event_id", "event_name"],
+        columns_rename_mapping={{"event_id": "id"}},
+        context_name="test_context",
+        skip_entity_validation_checks=True,
+        primary_entities=["event_id"],
+    )
+    """
+    assert textwrap.dedent(definition).strip() == textwrap.dedent(expected_definition).strip()
```
Comment (gee-senbong) [inline tests/unit/api/observation_table/test_utils.py:50]: Could consider to run the generated definition to make sure it is runnable.
```
local_vars = {}
exec(sdk_code, {}, local_vars)  # pylint: disable=exec-used
output = local_vars["output"]
```

**Review by gee-senbong — APPROVED** [review]

**Pull request merged**